app/services/CheckoutService.scala

```diff
@@ -2,25 +2,33 @@ package services
 
 import com.gu.identity.play.IdMinimalUser
 import com.gu.membership.salesforce.MemberId
+import com.gu.membership.zuora.soap.Zuora.SubscribeResult
 import com.typesafe.scalalogging.LazyLogging
 import model.SubscriptionData
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
+case class Subscription(memberId: MemberId, zuoraSubscription: SubscribeResult)
+
 class CheckoutService(identityService: IdentityService, salesforceService: SalesforceService) extends LazyLogging {
 
   def processSubscription(subscriptionData: SubscriptionData,
-                          idUser: Option[IdMinimalUser]): Future[Option[MemberId]] =
-    idUser.map(user => Future(Some(UserId(user.id))))
+                          idUser: Option[IdMinimalUser]): Future[Subscription] = {
+    def idUserF = idUser.map(user => Future(UserId(user.id)))
       .getOrElse {
-        logger.info(s"User does not have an Identity account, creating a Guest one.")
+      logger.info(s"User does not have an Identity account, creating a Guest one.")
+
+      identityService.registerGuest(subscriptionData.personalData)
+    }
+
+    for {
+      idUser <- idUserF
+      memberId <- salesforceService.createOrUpdateUser(subscriptionData.personalData, idUser)
+      subscription <- TouchpointBackend.Normal.zuoraService.createSubscription(memberId, subscriptionData)   //TODO when implementing test-users this requires updating to supply data to correct location
```

Comment:
We use TouchpointBackend.Normal.zuoraService as later we can update this to point to Normal (DEV / PROD) or Test (UAT). However this PR does not update Salesforce to distinguish between instances. @rtyley is it worth changing here or waiting until we implement test-users?